runPerftool.psm1

```diff
@@ -339,36 +380,52 @@ param(
             Invoke-Command -Session $sendPSSession -ScriptBlock $ScriptBlockRunToolCmd -ArgumentList $sendCmd
 
             # non blocking loop to check if the process made a clean exit
-            LogWrite "Waiting for $TimeoutValueBetweenCommandPairs seconds ..."
-            $timeout = new-timespan -Seconds $TimeoutValueBetweenCommandPairs
-            $sw = [diagnostics.stopwatch]::StartNew()
-            while ($sw.elapsed -lt $timeout){
+
+            # Calculate actual timeout value.
+            # For tools such as ntttcp, we may need to add additional #s for runtime, wu and cd times 
+            [int] $timeout = GetActualTimeOutValue -AdditionalTimeout $TimeoutValueBetweenCommandPairs -Line $sendCmd
+            LogWrite "Waiting for $timeout seconds ..."
+            $sw.Reset()
+            $sw.Start()
+            while (([math]::Round($sw.Elapsed.TotalSeconds,0)) -lt $timeout){
+
+                start-sleep -seconds 5
+
                 $checkRecvProcessExit = Invoke-Command -Session $recvPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
                 $checkSendProcessExit = Invoke-Command -Session $sendPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
 
                 if (($checkRecvProcessExit -eq $null)-and ($checkSendProcessExit -eq $null)){
+                    $cmdPairCompleted = $true
                     LogWrite "$Toolname exited on both Src and Dest machines"
                     break
                 }
-                start-sleep -seconds 5
             }
 
-        $checkRecvProcessExit = Invoke-Command -Session $recvPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
-        $checkSendProcessExit = Invoke-Command -Session $sendPSSession -ScriptBlock $CheckProcessExitScriptBlock -ArgumentList "$Toolname"
+            $sw.Stop()
+            
+            #Wait for disk I/O to be completed
+            Write-VolumeCache (get-location).Drive.Name
+            start-sleep -seconds 30
```

Comment:
There's several hardcoded constants.  Are these better saved of in a config file.  Consider the case where someone want's to modify them?  Would that case bee needed?  The config file could establish the reasonable time range to avoid misconfiguration.  Thoughts?